cpp/src/arrow/dataset/dataset.cc

```diff
@@ -0,0 +1,56 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+#include "arrow/dataset/dataset.h"
+
+#include <memory>
+#include <utility>
+
+#include "arrow/dataset/scanner.h"
+#include "arrow/util/stl.h"
+
+namespace arrow {
+namespace dataset {
+
+SimpleDataFragment::SimpleDataFragment(
+    std::vector<std::shared_ptr<RecordBatch>> record_batches)
+    : record_batches_(std::move(record_batches)) {}
+
+Status ScanTaskIteratorFromRecordBatches(
```

Comment:
This helper function could be folded into `Scan`, no?

Reply:
Yes, this got lost in a rebase (which renamed the function).

Reply:
The later function was initially more verbose thus the separation, I'll fold it.